capa/main.py

```diff
@@ -612,13 +612,18 @@ def get_rules_from_cli(args) -> RuleSet:
     raises:
       ShouldExitError: if the program is invoked incorrectly and should exit.
     """
+    enable_cache: bool = True
     try:
         if capa.helpers.is_running_standalone() and args.is_default_rules:
             cache_dir = get_default_root() / "cache"
         else:
             cache_dir = capa.rules.cache.get_default_cache_directory()
 
-        rules = capa.rules.get_rules(args.rules, cache_dir=cache_dir)
+        if capa.helpers.is_dev_environment():
+            # using the rules cache during development may result in unexpected errors, see #1898
+            enable_cache = capa.helpers.is_cache_newer_than_rule_code(cache_dir)
```

Comment:
```suggestion
            enable_cache = capa.helpers.is_cache_newer_than_rule_code(cache_dir)
            if not enable_cache:
                def ts_to_str(ts):
                    return datetime.fromtimestamp(ts).strftime("%Y-%m-%d %H:%M:%S")

            logger.warning(
                "not using cache: latest rule code file %s (%s) is newer than the latest rule cache file %s (%s)",
                latest_rule_code_file,
                ts_to_str(rule_code_timestamp),
                latest_cache_file,
                ts_to_str(cache_timestamp),
            )
            logger.debug("delete the cache file manually to use rule caching again")
```

Reply:
the timestamp values are not available here though

Reply:
oh i see. sorry did this through the GH web interface. guess we can keep some logic above, and the conclusion here. let me revise.

Reply:
revised both edits. please take a look @mr-tz